pkg/authorization/api/v1beta1/conversion.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"sort"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	newer "github.com/openshift/origin/pkg/authorization/api"
+)
+
+func init() {
+	err := api.Scheme.AddConversionFuncs(
+		func(in *Policy, out *newer.Policy, s conversion.Scope) error {
+			if err := s.Convert(&in.TypeMeta, &out.TypeMeta, 0); err != nil {
```

Comment:
Because we have virtual resources, we have to manage uniqueness ourselves, so having a map makes that easy.  It's also natural to want to access as a map in the code.

The default converter calls it, but with with a nil pointer.  Assigning the nil pointer does not propogate back to the containing object.